lib/matplotlib/axes/_axes.py

```diff
@@ -6411,11 +6411,18 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
         """
         Plot a histogram.
 
-        Compute and draw the histogram of *x*.  The return value is a tuple
-        (*n*, *bins*, *patches*) or ([*n0*, *n1*, ...], *bins*, [*patches0*,
-        *patches1*, ...]) if the input contains multiple data.  See the
-        documentation of the *weights* parameter to draw a histogram of
-        already-binned data.
+        Computes the histogram of the data in *x* (see ~numpy.histogram) and
+        draws the results. If the input is an array, then the return value is
+        a tuple (*n*, *bins*, *patches*); if the input is a sequence of arrays,
+        then the return value is a tuple
+        ([*n0*, *n1*, ...], *bins*, [*patches0*, *patches1*, ...]).
+
+        This method computes the histogram; therefore the data in *x* is
+        assumed to be unbinned and bins are computed when the *bins* parameter
+        is not set. If the histogram has already been computed, it can be
+        displayed using `~.bar` or `~.stairs`. This method can also be used to
+        draw a pre-computed histogram by passing in the counts and bins, as
+        described in the *weights* parameter documentation.
```

Comment:
I suggest to give more guidance 
```suggestion
        is not set. If you have already binned data, we recommend plotting
        them using `~.bar` or `~.stairs`. If you have a special need to plot
        already binned data using ``hist()``, you can trick it into doing so
        by fixing the bins and providing one value per bin::
        
            counts, bins = np.histogram(data)
            plt.hist(bins[:-1], bins, weights=counts)
```
And remove that part from the *weights* docs. It makes more sense in the general discussion here and should not be hidden in a parameter description.